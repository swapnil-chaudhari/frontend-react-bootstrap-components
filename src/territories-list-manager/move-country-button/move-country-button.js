import PropTypes from 'prop-types';
import React from 'react';
import ThreeArrowsIcon from './three-arrows-icon';
import './move-country-button.scss';

const MoveCountryButton = (
    {
        thisListType,
        theOtherListType,
        selectedCountriesNumber,
        onCountryMove,
        theOtherListLabel,
        forward,
        formatMessage,
        messages
    },
) => {
    const handleCountryMove = () =>
        onCountryMove(thisListType, theOtherListType);
    return (
        <div className="MoveCountryButton" onClick={ handleCountryMove }>
            <ThreeArrowsIcon
                reversed={ true }
                isHidden={ forward }
            />
            <div className="MoveCountryButton-text">
                <p>
                    { formatMessage(messages.moveSelectedLabel) }
                    { ` (${selectedCountriesNumber})` }
                </p>
                <p>
                    { formatMessage(messages.moveToLabel) }
                    <b>
                        { ` ${theOtherListLabel}` }
                    </b>
                </p>
            </div>
            <ThreeArrowsIcon
                reversed={ false }
                isHidden={ !forward }
            />
        </div>
    );
};

MoveCountryButton.propTypes = {
    thisListType: PropTypes.string.isRequired,
    theOtherListType: PropTypes.string.isRequired,
    selectedCountriesNumber: PropTypes.number.isRequired,
    onCountryMove: PropTypes.func,
    theOtherListLabel: PropTypes.string.isRequired,
    forward: PropTypes.bool.isRequired,
    formatMessage: PropTypes.func.isRequired,
    messages: PropTypes.object.isRequired
};

export default MoveCountryButton;
