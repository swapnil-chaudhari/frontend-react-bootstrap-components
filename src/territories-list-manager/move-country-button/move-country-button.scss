@import "src/styles/variables";

.MoveCountryButton {
    align-items: center;
    display: flex;
    justify-content: center;

    &-icons {
        flex-grow: 0;
    }

    &-reverse {
        fill: $white;
        transform: rotateY(180deg);
    }

    &-text {
        flex-grow: 3;
        font-size: 14px;
    }
}
