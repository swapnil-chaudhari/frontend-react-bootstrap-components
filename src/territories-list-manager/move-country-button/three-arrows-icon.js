import PropTypes from 'prop-types';
import React from 'react';
import cx from 'classnames';
import ThreeRightDirectedArrows from 'src/icons/three-right-directed-arrows';

const ThreeArrowsIcon = ({ reversed, isHidden }) => {
    const iconsClasses = cx({
        'MoveCountryButton-icons': true,
        'MoveCountryButton-reverse': reversed,
        hide: isHidden
    });

    return (
        <div className={ iconsClasses }>
            <ThreeRightDirectedArrows />
        </div>
    );
};

ThreeArrowsIcon.propTypes = {
    reversed: PropTypes.bool.isRequired,
    isHidden: PropTypes.bool.isRequired,
};

export default ThreeArrowsIcon;
