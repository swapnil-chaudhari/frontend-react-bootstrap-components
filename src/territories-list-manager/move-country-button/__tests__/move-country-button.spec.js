import React from 'react';
import renderShallow from 'render-shallow';
import { default as MoveCountryButton } from '../move-country-button';
import noop from 'src/utils/noop';


describe('<MoveCountryButton>', () => {
    describe('when it renders', () => {
        const props = {
            thisListType: 'foo',
            theOtherListType: 'bar',
            selectedCountriesNumber: 1,
            onCountryMove: noop,
            theOtherListLabel: 'The Other List Label',
            forward: true,
            formatMessage: msg => msg,
            messages: {
                moveSelectedLabel: 'move selected label',
                moveToLabel: 'move to label'
            }
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<MoveCountryButton { ...props } />).output;
        });

        test('renders button with correct data', () => {
            expect(component).toMatchSnapshot();
        });
    });

});
