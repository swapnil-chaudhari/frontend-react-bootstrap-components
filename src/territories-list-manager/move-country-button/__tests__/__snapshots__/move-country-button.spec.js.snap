// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoveCountryButton> when it renders renders button with correct data 1`] = `
<div
  className="MoveCountryButton"
  onClick={[Function]}
>
  <ThreeArrowsIcon
    isHidden={true}
    reversed={true}
  />
  <div
    className="MoveCountryButton-text"
  >
    <p>
      move selected label
       (1)
    </p>
    <p>
      move to label
      <b>
         The Other List Label
      </b>
    </p>
  </div>
  <ThreeArrowsIcon
    isHidden={false}
    reversed={false}
  />
</div>
`;
